components/form/FAutocompleteInput.vue

```diff
@@ -306,40 +305,51 @@ const {
   hint,
   value: fieldValue,
   validate,
-} = useFieldWithValidation<string | number>(props, {
+} = useFieldWithValidation<unknown>(props, {
   validateOnMount: props?.validateOnMount,
   rules: [isValidMatch],
 });
 
-const {
-  handleBlur,
-  handleChange,
-  handleFocus: _handleFocus,
-  handleInput,
-} = useInputEventBindings(validate, props.validationTrigger, emit);
+function handleBlur(e: Event, closeMenu: () => void) {
+  closeMenu();
+  emit('blur', e);
+
+  // Wait for the menu to fade out before clearing the input to avoid seeing options change
+  setTimeout(() => {
+    if (!currentOptionMatched.value) {
+      inputValue.value = '';
+      fieldValue.value = undefined;
+    }
+  }, 300);
+}
 
 function handleFocus(e: Event, openMenu: () => void) {
   openMenu();
-  _handleFocus(e);
+  emit('focus', e);
+}
+
+function handleInput(e: InputEvent) {
+  emit('input', e);
+  currentOptionMatched.value = undefined;
 }
 
-watch(fieldValue, newValue => {
-  // Fire an event whenever field value change to handle eventual API calls
-  emit('value-change', newValue);
+watch(inputValue, newValue => {
+  // Fire an event whenever input value change to handle eventual API calls
+  emit('input-value', newValue);
 });
 
 function isValidMatch() {
   return !!currentOptionMatched.value;
 }
 
-function handleSelectOption(optionValue: string | number | null) {
-  inputIsActive.value = false;
-
+function handleSelectOption(optionValue: unknown) {
   fieldValue.value = optionValue;
-  currentOptionMatched.value = props.options.find(
-    option => fieldValue.value === option.value
+  currentOptionMatched.value = props.options.find(option =>
+    equal(optionValue, option.value)
   );
 
+  inputValue.value = (currentOptionMatched.value as FMenuOption).label;
```

Comment:
Why do you need to cast `currentOptionMatched.value` as `FMenuOption` ?
I think `props.options` already has the type `FMenuOption[]`, so the only problem here is that the `find` may return `undefined. Hence you should be able just to do:
```suggestion
  inputValue.value = currentOptionMatched.value?.label;
```
and it's *safer*.

Reply:
The reason I didn't do it was because `composeSearchRegex()` didn't accept undefined values, but I fixed it. Thanks !